### PR TITLE
[ASSIST-425] & [ASSIST-369]

### PR DESCRIPTION
**Pull request by Orrison**
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/425/
https://engage.canyongbs.com/workgroups/group/3/tasks/task/view/369/

- Changes Student and Prospects `full` column to `full_name` to avoid Postgres conflict
- Creates a unifying interface `Educatable`, to enforce and use methods for shared information between Students and Prospects when they are used interchangeably throughout the system
- Fixes the issue preventing us from using the IDE helper Model `@mixin` and replaces all huge Docblocks with `@mixin`. This will not only reduce the line size of our Models but also allow us to override/set Docblock properties when needed
- Deletes old Models and code

### CODE REVIEW

File: app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/CreateProspect.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function form(Form $form): Form
                     ->translateLabel()
                     ->required()
                     ->string(),
-                TextInput::make('full')
+                TextInput::make('full_name')
```
Comment (dgoetzit) [inline app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/CreateProspect.php:51]:
```suggestion
                TextInput::make(Prospect::displayNameKey())
```

File: app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ViewProspect.php [hunk 2]
```diff
@@ -35,7 +35,7 @@ public function infolist(Infolist $infolist): Infolist
                         TextEntry::make('last_name')
                             ->label('Last Name')
                             ->translateLabel(),
-                        TextEntry::make('full')
+                        TextEntry::make('full_name')
```
Comment (dgoetzit) [inline app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ViewProspect.php:38]:
```suggestion
                        TextEntry::make(Prospect::displayNameKey())
```

**Review by dgoetzit — CHANGES_REQUESTED** [review]
Looks good! Just a few suggestions for consistency.

File: app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/EditProspect.php [hunk 3]
```diff
@@ -49,7 +49,7 @@ public function form(Form $form): Form
                     ->translateLabel()
                     ->required()
                     ->string(),
-                TextInput::make('full')
+                TextInput::make('full_name')
```
[the remark this answers is not in the document]
Comment (Orrison) [reply app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/EditProspect.php:52]: Fixed

File: app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/CreateProspect.php (hunk 1 above)
Comment (Orrison) [reply app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/CreateProspect.php:51]: Fixed

File: app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ViewProspect.php (hunk 2 above)
Comment (Orrison) [reply app-modules/prospect/src/Filament/Resources/ProspectResource/Pages/ViewProspect.php:38]: Fixed

**Review by dgoetzit — APPROVED** [review]

**Pull request merged**